hibernate-core/src/main/java/org/hibernate/internal/util/QuotingHelper.java

```diff
@@ -70,13 +70,24 @@ public static String unquoteIdentifier(String text) {
 	}
 
 	public static String unquoteStringLiteral(String text) {
+		return unquoteString( text, false );
+	}
+
+	public static String unquoteJavaStringLiteral(String text) {
+		return unquoteString( text, true );
+	}
+
+	public static String unquoteString(String text, boolean unescape) {
 		assert text.length() > 1;
+		int start = 0;
```

Comment:
I'd prefer passing the start index explicitly.

```suggestion
		return unquoteString( text, false, 0 );
	}

	public static String unquoteJavaStringLiteral(String text) {
		final char firstChar = text.charAt( 0 );
		return unquoteString( text, true, firstChar == 'j' || firstChar == 'J' ? 1 : 0 );
	}

	public static String unquoteString(String text, boolean unescape, int start) {
		assert text.length() > 1;
```